docs/content/envoy-external-authorization.md

```diff
@@ -0,0 +1,334 @@
+
+> In typical deployments the policy would either be built into the OPA container
+> image or it would fetched dynamically via the [Bundle
+> API](https://www.openpolicyagent.org/docs/latest/bundles/). ConfigMaps are
+> used in this tutorial for test purposes.
+
+### 4. Create App Deployment with OPA and Envoy sidecars
+
+Our deployment contains a sample Go app which provides information about employees in a company. It exposes a `/people` endpoint to `get` and `create` employees. More information can on the app be found [here](https://github.com/ashutosh-narkar/go-test-server).
+
+Save the deployment as **deployment.yaml**:
+
+```yaml
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: example-app
+  labels:
+    app: example-app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: example-app
+  template:
+    metadata:
+      labels:
+        app: example-app
+    spec:
+      initContainers:
+        - name: proxy-init
+          image: openpolicyagent/proxy_init:v2
+          args: ["-p", "8000", "-u", "1111"]
+          securityContext:
+            capabilities:
+              add:
+              - NET_ADMIN
+            runAsNonRoot: false
+            runAsUser: 0
+      containers:
+        - name: app
+          image: openpolicyagent/demo-test-server:v1
+          ports:
+            - containerPort: 8080
+        - name: envoy
+          image: envoyproxy/envoy:v1.10.0
+          securityContext:
+            runAsUser: 1111
+          volumeMounts:
+          - readOnly: true
+            mountPath: /config
+            name: proxy-config
+          args:
+          - "envoy"
+          - "--config-path"
+          - "/config/envoy.yaml"
+        - name: opa
+          # Note: openpolicyagent/opa:edge-istio is created by retagging
+          # the latest released image of OPA-Istio.
+          image: openpolicyagent/opa:{{< current_opa_istio_docker_version >}}
+          securityContext:
+            runAsUser: 1111
+          volumeMounts:
+          - readOnly: true
+            mountPath: /config
+            name: opa-config
+          - readOnly: true
+            mountPath: /policy
+            name: opa-policy
+          args:
+          - "--plugin-dir=/app"
+          - "run" 
+          - "--server"
+          - "--config-file=/config/opa.yaml"
+          - "--ignore=.*"
+          - "/policy/policy.rego"
+      volumes:
+        - name: proxy-config
+          configMap:
+            name: proxy-config
+        - name: opa-config
+          configMap:
+            name: opa-config
+        - name: opa-policy
+          configMap:
+            name: opa-policy
+```
+
+```bash
+kubectl apply -f deployment.yaml
+```
+
+> The `proxy-init` container installs iptables rules to redirect all container traffic through the Envoy proxy sidecar. More information can be found [here](https://github.com/tsandall/minimal-opa-envoy-example/tree/master/proxy_init).
```

Comment:
We should move this out of my personal GitHub repo. I think we could just stick the script in the contrib repo. The contrib repo also has Travis CI hooked up so we could automatically build & publish the image when the script changes.

Reply:
Nit: also, this script only redirects incoming traffic via Envoy. Outgoing traffic bypasses Envoy completely.